backend/src/test/scala/com/softwaremill/bootzooka/test/TestHelpersWithDb.scala

```diff
@@ -4,20 +4,22 @@ import com.softwaremill.bootzooka.common.sql.SqlDatabase
 import com.softwaremill.bootzooka.email.application.{DummyEmailService, EmailTemplatingEngine}
 import com.softwaremill.bootzooka.user.application.{UserDao, UserService}
 import com.softwaremill.bootzooka.user.domain.User
+import org.scalatest.AsyncFlatSpec
 import org.scalatest.concurrent.ScalaFutures
-
 import scala.concurrent.ExecutionContext
 
 trait TestHelpersWithDb extends TestHelpers with ScalaFutures {
 
+  this: AsyncFlatSpec =>
+
   lazy val emailService = new DummyEmailService()
   lazy val emailTemplatingEngine = new EmailTemplatingEngine
-  lazy val userDao = new UserDao(sqlDatabase)
-  lazy val userService = new UserService(userDao, emailService, emailTemplatingEngine)
+  lazy val initEc = ExecutionContext.global
 
-  def sqlDatabase: SqlDatabase
+  lazy val userDao = new UserDao(sqlDatabase)(initEc)
```

Comment:
Ah, right ... this limitation kind of defeats the purpose of `beforeEach`. There is no variant which would allow you to use the built-in EC? Otherwise, we have to leave this as it is here